test/api/test_workflows.py

```diff
@@ -3047,6 +3047,37 @@ def test_run_replace_params_by_tool(self):
         self.__assert_lines_hid_line_count_is(history_id, 2, 5)
         self.__assert_lines_hid_line_count_is(history_id, 3, 5)
 
+    @skip_without_tool("random_lines1")
+    def test_integer_param_replacement(self):
+        workflow_id = self._upload_yaml_workflow("""
+class: GalaxyWorkflow
+inputs:
+  test_input: data
+steps:
+  first:
+    tool_id: random_lines1
+    state:
+      num_lines: ${numlines}
```

Comment:
```suggestion
      num_lines: {numlines}
```

Reply:
Ahh, is this a yaml-format specific thing?  When you download a workflow .ga it's the same `${numlines}` as you'd use in the UI.

Reply:
just to recap, the way you're doing it is correct, I got confused by another test